sdk/metrics/src/jmhBasedTest/java/io/opentelemetry/sdk/metrics/internal/state/InstrumentGarbageCollectionBenchmark.java

```diff
@@ -51,37 +53,46 @@
  */
 @BenchmarkMode(Mode.SingleShotTime)
 @OutputTimeUnit(TimeUnit.NANOSECONDS)
-@Measurement(iterations = 20, batchSize = 100)
+@Measurement(iterations = 10, batchSize = 10)
 @Warmup(iterations = 10, batchSize = 10)
 @Fork(1)
-public class AsynchronousMetricStorageGarbageCollectionBenchmark {
+public class InstrumentGarbageCollectionBenchmark {
 
   @State(value = Scope.Benchmark)
-  @SuppressWarnings("SystemOut")
   public static class ThreadState {
     private final int cardinality;
-    private final int countersCount;
+    private final int instrumentCount;
+    @Param public TestInstrumentType testInstrumentType;
     @Param public AggregationTemporality aggregationTemporality;
     @Param public MemoryMode memoryMode;
     SdkMeterProvider sdkMeterProvider;
     private final Random random = new Random();
     List<Attributes> attributesList;
+    private TestInstrumentsState testInstrumentsState;
+    private InstrumentTester instrumentTester;
 
     /** Creates a ThreadState. */
     @SuppressWarnings("unused")
     public ThreadState() {
       cardinality = 1000;
-      countersCount = 10;
+      instrumentCount = 10;
     }
 
     @SuppressWarnings("SpellCheckingInspection")
     @Setup
-    public void setup() {
+    public void setup()
+        throws NoSuchMethodException,
+            InvocationTargetException,
+            InstantiationException,
+            IllegalAccessException {
+      instrumentTester =
+          testInstrumentType.instrumentTesterClass.getDeclaredConstructor().newInstance();
```

Comment:
#nit: If you change `instrumentTesterClass` from `Class<? extends InstrumentTester>` to `Supplier<InstrumentTester>`, and configure each element in the `TestInstrumentType` enum with a factory method for creating new instances of `InstrumentTester`:

```
public enum TestInstrumentType {
  ASYNC_COUNTER(AsyncCounterTester::new),
  EXPONENTIAL_HISTOGRAM(ExponentialHistogramTester::new);

  final Supplier<InstrumentTester> instrumentTesterSupplier;

  TestInstrumentType(Supplier<InstrumentTester> instrumentTesterSupplier) {
    this.instrumentTesterSupplier = instrumentTesterSupplier;
  }
 
  ..
```

Reply:
Thanks. Much better.
I did it with an abstract method since error prone recommended it, and it does seem cleaner.

I also added the `ProfileBenchmark` and linked to it from the test